reference/employer/openapi.yaml

```diff
@@ -0,0 +1,1083 @@
+      summary: Individual
+      operationId: get-individual
+      tags:
+        - Employer
+      requestBody:
+        $ref: '#/components/requestBodies/Individual-Ids'
+  /employer/employment:
+    post:
+      description: Read individual employment and income data
+      parameters:
+        - $ref: '#/components/parameters/API-Version'
+      responses:
+        '200':
+          description: Employment data
+          content:
+            application/json:
+              examples:
+                Success:
+                  value:
+                    responses:
+                      - individual_id: 5d0b10a1-a09a-430f-81f1-20be735dc5e9
+                        code: 200
+                        body:
+                          id: 5d0b10a1-a09a-430f-81f1-20be735dc5e9
+                          first_name: Jane
+                          middle_name: null
+                          last_name: Doe
+                          title: Customer Support
+                          manager:
+                            id: c205b3fa-b626-4346-bf0f-ca065ab88d31
+                          department:
+                            name: Product
+                          employment:
+                            type: employee
+                            subtype: full_time
+                          start_date: '2017-06-30'
+                          end_date: null
+                          is_active: true
+                          location:
+                            line1: 12 Bird Drive
+                            line2: null
+                            city: Palo Alto
+                            state: CA
+                            postal_code: '94301'
+                            country: US
+                          income:
+                            unit: yearly
+                            amount: 10000000
+                            currency: usd
+                            effective_date: '2019-03-01'
+                          income_history:
+                            - unit: yearly
+                              amount: 9000000
+                              currency: usd
+                              effective_date: '2018-05-15'
+                            - unit: yearly
+                              amount: 8000000
+                              currency: usd
+                              effective_date: '2017-06-30'
+              schema:
+                type: object
+                properties:
+                  responses:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        individual_id:
+                          type: string
+                        code:
+                          type: integer
+                        body:
+                          type: object
+                          properties:
+                            id:
+                              type: string
```

Comment:
for employment, first_name, last_name, employment, location, start_date, end_date and income are nullable as well! 

For employment, employment.type is not nullable while subtype is nullable, _I think_.

Reply:
following our controller validation
```
const responseSchema = Joi.object({
  id: Joi.string().guid(),
  work_id: Joi.any().allow(null).optional(),
  first_name: Joi.string(),
  last_name: Joi.string(),
  middle_name: Joi.string().allow(null).allow(''),
  title: Joi.string().allow(null),
  manager: Joi.object({
    id: Joi.string().guid(),
  }).allow(null),
  department: Joi.object({
    name: Joi.string(),
  }).allow(null),
  employment: Joi.object({
    type: Joi.string().valid('employee', 'contractor'),
    subtype: Joi.string()
      .valid(
        'full_time',
        'part_time',
        'intern',
        'temp',
        'seasonal',
        'individual_contractor'
      )
      .allow(null),
  }),
  start_date: commonSchemas.date,
  end_date: commonSchemas.date,
  is_active: Joi.boolean().allow(null),
  income: incomeSchema,
  income_history: Joi.array().items(incomeSchema).allow(null),
  location: commonSchemas.location,
  teams: Joi.array()
    .items(
      Joi.object({
        name: Joi.string(),
      })
    )
    .allow(null)
    .optional(),
});
```

Reply:
for location and income, you'll have to check the actual reference object itself.

Reply:
The income ref doesn't seem nullable, rest are!